tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -92,65 +114,106 @@ class StorageSetNetworkRulesAction(AzureBaseAction):
                   value: 0
 
             actions:
-                - type: set-network-rules
-                  default-action: Deny
-                  bypass: [Logging, Metrics]
+                - type: set-firewall-rules
+                  bypass-rules:
+                      - Logging
+                      - Metrics
                   ip-rules:
-                      - ip-address-or-range: 11.12.13.14
-                      - ip-address-or-range: 21.22.23.24
+                      - 11.12.13.0/16
+                      - 21.22.23.24
                   virtual-network-rules:
-                      - virtual-network-resource-id: <subnet_resource_id>
-                      - virtual-network-resource-id: <subnet_resource_id>
+                      - <subnet_resource_id>
+                      - <subnet_resource_id>
 
     """
 
     schema = type_schema(
-        'set-network-rules',
-        required=['default-action'],
+        'set-firewall-rules',
+        required=[],
         **{
-            'default-action': {'enum': ['Allow', 'Deny']},
-            'bypass': {'type': 'array', 'items': {'enum': ['AzureServices', 'Logging', 'Metrics']}},
-            'ip-rules': {
-                'type': 'array',
-                'items': {'ip-address-or-range': {'type': 'string'}}
-            },
-            'virtual-network-rules': {
-                'type': 'array',
-                'items': {'virtual-network-resource-id': {'type': 'string'}}
-            }
+            'default-action': {'enum': ['Allow', 'Deny'], "default": 'Deny'},
+            'append': {'type': 'boolean', "default": False},
+            'bypass-rules': {'type': 'array', 'items': {
+                'enum': ['AzureServices', 'Logging', 'Metrics']}},
+            'ip-rules': {'type': 'array', 'items': {'type': 'string'}},
+            'virtual-network-rules': {'type': 'array', 'items': {'type': 'string'}}
         }
     )
 
-    def _prepare_processing(self,):
+    def __init__(self, data, manager=None):
+        super(StorageSetNetworkRulesAction, self).__init__(data, manager)
+        self._log = logging.getLogger('custodian.azure.storage')
+        self.rule_limit = 200
```

Comment:
Can we add this number to the docs?

Reply:
Yeah! And I figure later each subclass will set this value.